presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteAggregationIfToFilter.java

```diff
@@ -74,7 +77,10 @@
  * <p>
  * Note that unwrapping the IF expression in the aggregate might cause issues if the true branch return errors for rows not matching the filters. For example:
  * 'IF(CARDINALITY(array) > 0, array[1]))'
- * Session property AGGREGATION_IF_TO_FILTER_REWRITE_STRATEGY and canUnwrapIf() control whether to enable IF unwrapping.
+ * Session property AGGREGATION_IF_TO_FILTER_REWRITE_STRATEGY and canUnwrapIf() control whether to enable IF unwrapping:
+ * 1. If the strategy is FILTER_WITH_IF, then keep the IF expression.
+ * 2. If it is certain that the IF expression can be unwrapped safe, then do so.
+ * 3. If the strategy is UNWRAP_IF, then unwrap the IF expression after it passes the remaining checks.
```

Comment:
Mention that this is an unsafe mode and the checks are not exhaustive. There may be cases where unwrapping the if could lead to errors.